api/proto/teleport/legacy/types/types.proto

```diff
@@ -5120,9 +5129,16 @@ message PluginBootstrapCredentialsV1 {
   oneof credentials {
     PluginOAuth2AuthorizationCodeCredentials oauth2_authorization_code = 1;
     PluginBearerTokenCredentials bearer_token = 2;
+    PluginIdSecretCredential id_secret = 3;
   }
 }
 
+// PluginIdSecretCredential can be OAuth2-like client_id and client_secret or username and password.
```

Comment:
In what cases will we accept OAuth2 `client_id` and `client_secret` from the user? Will this be how we handle Mattermost etc., where we can not have one predefined Client ID, owned by us?